packages/dd-trace/src/profiling/profilers/wall.js

```diff
@@ -76,6 +76,43 @@ function getWebTags (startedSpans, i, span) {
   return memoize(null)
 }
 
+let channelsActivated = false
+function ensureChannelsActivated () {
+  if (channelsActivated) return
+
+  const { AsyncLocalStorage, createHook } = require('async_hooks')
+  const shimmer = require('../../../../datadog-shimmer')
+
+  createHook({ before: () => beforeCh.publish() }).enable()
+
+  shimmer.wrap(AsyncLocalStorage.prototype, 'enterWith', function (original) {
+    return function (...args) {
+      const retVal = original.apply(this, args)
+      enterCh.publish()
+      return retVal
+    }
+  })
+
+  // TODO this will change in the future. It won't be experimental forever.
+  const needsWrappedCb =
+    !process.execArgv.includes('--experimental-async-context-frame')
```

Comment:
If a version ships with it enabled by default this will break. Any other way to detect that the new code is running?

Reply:
Apart from version ranges (which we won't know in advance), the only other thing would be that the legacy implementation has two (undocumented) properties hanging off it. Would you rather I check for one or both of those properties instead?